src/graphic/animation/nonpacked_animation.cc

```diff
@@ -0,0 +1,238 @@
+	return frames.at(0)->width();
+}
+int NonPackedAnimation::NonPackedMipMapEntry::height() const {
+	return frames.at(0)->height();
+}
+
+/*
+==============================================================================
+
+NonPackedAnimation IMPLEMENTATION
+
+==============================================================================
+*/
+
+NonPackedAnimation::NonPackedAnimation(const LuaTable& table, const std::string& basename)
+   : Animation(table) {
+	try {
+		// Get image files
+		if (table.has_key("pictures")) {
+			// TODO(GunChleoc): Old code - remove this option once conversion has been completed
+			mipmaps_.insert(std::make_pair(
+			   1.0f, std::unique_ptr<NonPackedMipMapEntry>(
+			            new NonPackedMipMapEntry(table.get_table("pictures")->array_entries<std::string>()))));
+		} else {
+			if (basename.empty() || !table.has_key("directory")) {
+				throw Widelands::GameDataError(
+				   "Animation did not define both a basename and a directory for its image files");
+			}
+			add_available_scales(basename, table.get_string("directory"));
+		}
+
+		// Frames
+		const NonPackedMipMapEntry& first = dynamic_cast<const NonPackedMipMapEntry&>(*mipmaps_.begin()->second.get());
+		nr_frames_ = first.image_files.size();
+		if (table.has_key("fps") && nr_frames_ == 1) {
+				throw Widelands::GameDataError("Animation with one picture %s must not have 'fps'",
+				                               first.image_files.front().c_str());
+		}
+
+		if (representative_frame() < 0 || representative_frame() > nr_frames_ - 1) {
+			throw wexception("Animation has %d as its representative frame, but the frame indices "
+			                 "available are 0 - %d",
+			                 representative_frame(), nr_frames_ - 1);
+		}
+
+		// Perform some checks to make sure that the data is complete and consistent
+		const bool should_have_playercolor = mipmaps_.begin()->second->has_playercolor_masks;
+		for (const auto& mipmap : mipmaps_) {
+			const NonPackedMipMapEntry& nonpacked_mipmap = dynamic_cast<const NonPackedMipMapEntry&>(*mipmap.second.get());
+			if (nonpacked_mipmap.image_files.size() != nr_frames_) {
+				throw Widelands::GameDataError(
+				   "Mismatched number of images for different scales in animation table: %" PRIuS
+				   " vs. %u at scale %.2f",
+				   nonpacked_mipmap.image_files.size(), nr_frames_, static_cast<double>(mipmap.first));
+			}
+			if (nonpacked_mipmap.has_playercolor_masks != should_have_playercolor) {
+				throw Widelands::GameDataError(
+				   "Mismatched existence of player colors in animation table for scales %.2f and %.2f",
+				   static_cast<double>(mipmaps_.begin()->first), static_cast<double>(mipmap.first));
+			}
+		}
+		if (mipmaps_.count(1.0f) != 1) {
+			throw Widelands::GameDataError(
+			   "All animations must provide images for the neutral scale (1.0)");
+		}
+	} catch (const LuaError& e) {
+		throw Widelands::GameDataError("Error in animation table: %s", e.what());
```

Comment:
Please add the basename to the message, such errors are always horrible to debug

Reply:
I'll add the information to map_object.cc. Sample output:

```
Could not read world information: [../src/scripting/lua_errors.cc:22] [../src/scripting/lua_errors.cc:22] Error loading animation 'idle' for map object 'pebble1': NOCOM Trigger test exception
stack traceback:
	....
```